src/actions/actions.ts

```diff
@@ -2246,6 +2249,48 @@ class MoveToMatchingBracket extends BaseMovement {
   }
 }
 
+abstract class MoveInsideCharacter extends MoveToMatchingBase {
+  modes = [ModeName.Normal, ModeName.Visual, ModeName.VisualLine];
+  protected charToMatch: string;
+
+  public async execAction(position: Position, vimState: VimState): Promise<Position | IMovement> {
+    const text = TextEditor.getLineAt(position).text;
+    const pairedCharacter = this.pairings[this.charToMatch]
+    const charToClose = pairedCharacter ? pairedCharacter.match : this.charToMatch;
+
+    // First, search backwards for the opening character of the sequence
+    let start = -1;
+    for (let i = position.character; i >= 0; i--) {
+      if (text[i] == this.charToMatch) {
+        start = i;
+        break;
+      }
+    }
+    const startPos = new Position(position.line, start + 1);
+    const endPos = this.nextBracket(startPos, this.charToMatch, { match: charToClose, nextMatchIsForward: true }, false);
+    if (start == -1 || endPos.character == start + 1 || start > position.character || endPos.character < position.character ) {
+      return position;
+    }
+
+    return {
+      start : startPos,
+      stop  : endPos,
+    };
+  }
+}
+
+@RegisterAction
+class MoveIParentheses extends MoveInsideCharacter {
+  keys = ["i", "("];
+  charToMatch = "(";
+}
+
+@RegisterAction
+class MoveIDoubleQuote extends MoveInsideCharacter {
+  keys = ["i", "\""];
```

Comment:
I am actually okay with this. I'd prefer to explicitly list out every key combination that exists, even though it's slightly less DRY than we'd otherwise want.